Vesper-Agents-Suite/Fund/src/agent.ts

```diff
@@ -28,8 +35,13 @@ function provideHandleFunction(web3: Web3): HandleBlock {
       const pool = value[0];
       const totalValue = new BigNumber(value[1]);
       const tokenHere = new BigNumber(value[2]);
-
-      if (tokenHere.isGreaterThan(totalValue.multipliedBy(0.2))) {
+      const [success, time] = tracker.tryGetLastTime(pool);
+      if (!success) {
+        // set this block as the time to start tracking the pool fund
+        tracker.update(pool, blockEvent.block.timestamp);
+      };
+      const elapsed: number = blockEvent.block.timestamp - time;
+      if (elapsed >= timeThreshold && tokenHere.isGreaterThan(totalValue.multipliedBy(0.35))) {
```

Comment:
I think you should update the tracker inside this if:
```suggestion
      if (elapsed >= timeThreshold && tokenHere.isGreaterThan(totalValue.multipliedBy(0.35))) {
        tracker.update(pool, blockEvent.block.timestamp);
```